packages/block-editor/src/store/selectors.js

```diff
@@ -1651,6 +1651,37 @@ const isBlockVisibleInTheInserter = (
 	return true;
 };
 
+/**
+ * Determines if a container (clientId) allows insertion of blocks, considering contentOnly mode restrictions.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId The client ID of the container block.
+ *
+ * @return {boolean} Whether the container allows insertion.
+ */
+export function isContainerInsertableToInContentOnlyMode( state, clientId ) {
```

Comment:
I think 99% of the new selectors should start as private; we cannot remove anything that goes public in the release. Since this is only used by other selectors for the moment, I think this one can be an internal helper/selector.

Reply:
Ah yes, I was about to ask for thoughts on that! Makes sense to make it private for now.